crates/edr_provider/src/data.rs

```diff
@@ -1922,6 +1914,141 @@ impl<LoggerErrorT: Debug, TimerT: Clone + TimeSinceEpoch> ProviderData<LoggerErr
         Ok(transaction_hash)
     }
 
+    /// Returns the block header and the transactions from the block that
+    /// contains the transaction for debug trace transactions.
+    fn block_data_for_debug_trace_transaction(
+        &self,
+        transaction_hash: &B256,
+    ) -> Result<(Header, Vec<Signed>), ProviderError<LoggerErrorT>> {
+        // This is a hack to make `debug_traceTransaction` return a helpful error
```

Comment:
```suggestion
        // HACK: This is a hack to make `debug_traceTransaction` return a helpful error
```

Nit: I find it helpful to have the `HACK:` prefix to make it easily searchable when looking for tech debt.

Reply:
Fixed in https://github.com/NomicFoundation/edr/pull/606/commits/0e66f07f3fce6f2646fb479b538d003caf5ab95e